crates/sui-cluster-test/src/cluster.rs

```diff
@@ -0,0 +1,157 @@
+                String::from(DEVNET_FAUCET_ADDR),
+                String::from(DEVNET_FULLNODE_ADDR),
+            ),
+            Env::Staging => (
+                String::from(STAGING_GATEWAY_ADDR),
+                String::from(STAGING_FAUCET_ADDR),
+                String::from(STAGING_FULLNODE_ADDR),
+            ),
+            Env::CustomRemote => (
+                options
+                    .gateway_address
+                    .clone()
+                    .expect("Expect 'gateway_address' for Env::Custom"),
+                options
+                    .faucet_address
+                    .clone()
+                    .expect("Expect 'faucet_address' for Env::Custom"),
+                options
+                    .fullnode_address
+                    .clone()
+                    .expect("Expect 'fullnode_address' for Env::Custom"),
+            ),
+            Env::NewLocal => unreachable!("NewLocal shouldn't use RemoteRunningCluster"),
+        };
+        // TODO: test connectivity before proceeding?
+        Ok(Self {
+            rpc_url,
+            faucet_url,
+            fullnode_url,
+        })
+    }
+    fn rpc_url(&self) -> &str {
+        &self.rpc_url
+    }
+    fn fullnode_url(&self) -> &str {
+        &self.fullnode_url
+    }
+    fn faucet_url(&self) -> Option<&str> {
+        Some(&self.faucet_url)
+    }
+
+    fn user_key(&self) -> KeyPair {
+        get_key_pair().1
+    }
+}
+
+/// Represents a local Cluster which starts per cluster test run.
+pub struct LocalNewCluster {
+    test_network: TestNetwork,
+    fullnode_url: String,
+}
+
+impl LocalNewCluster {
+    fn swarm(&self) -> &Swarm {
+        &self.test_network.network
+    }
+}
+
+#[async_trait]
+impl Cluster for LocalNewCluster {
+    async fn start(_options: &ClusterTestOpt) -> Result<Self, anyhow::Error> {
+        let genesis_config = GenesisConfig::for_local_testing();
+        let test_network = start_rpc_test_network(Some(genesis_config), Some(1))
+            .await
+            .unwrap_or_else(|e| panic!("Failed to start a local network, e: {e}"));
+        let fullnode: &Node = test_network
+            .network
+            .fullnodes()
+            .next()
+            .expect("Expect one fullnode");
+        let fullnode_url = format!("http://{}", fullnode.json_rpc_address());
+
+        // Let nodes connect to one another
+        tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
+
+        // TODO: test connectivity before proceeding?
+        Ok(Self {
+            test_network,
+            fullnode_url,
+        })
+    }
+
+    fn rpc_url(&self) -> &str {
+        &self.test_network.rpc_url
+    }
+
+    fn fullnode_url(&self) -> &str {
+        &self.fullnode_url
+    }
+
+    // For now, a local cluster does not have faucet
+    fn faucet_url(&self) -> Option<&str> {
+        None
+    }
+
+    fn user_key(&self) -> KeyPair {
+        self.swarm().config().account_keys[0].copy()
+    }
```

Comment:
We should probably move more towards only creating 1 account at startup that is the "treasury" account and then tests can request funds similar to how we use faucet.

Reply:
agree